File: extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/PanacheMongoRepository.java
package io.quarkus.mongodb.panache;

import org.bson.types.ObjectId;

/**
 * Represents a Repository for a specific type of entity {@code Entity}, with an ID type
 * of {@code ObjectId}. Implementing this repository will gain you the exact same useful methods
 * that are on {@link PanacheMongoEntityBase}. If you have a custom ID strategy, you should
 * implement {@link PanacheMongoRepositoryBase} instead.
 *
 * @param <Entity> The type of entity to operate on
 */
public interface PanacheMongoRepository<Entity> extends PanacheMongoRepositoryBase<Entity, ObjectId> {

}

File: extensions/panache/mongodb-panache-kotlin/runtime/src/main/kotlin/io/quarkus/mongodb/panache/kotlin/PanacheMongoRepository.kt
package io.quarkus.mongodb.panache.kotlin

import org.bson.types.ObjectId

/**
 * Represents a Repository for a specific type of entity `Entity`, with an ID type
 * of `ObjectId`. Implementing this repository will gain you the exact same useful methods
 * that are on [PanacheMongoEntityBase]. If you have a custom ID strategy, you should
 * implement [PanacheMongoRepositoryBase] instead.
 *
 * @param Entity The type of entity to operate on
 */
interface PanacheMongoRepository<Entity: Any> : PanacheMongoRepositoryBase<Entity, ObjectId>